Client/Frontend/UserContent/UserScripts/FarblingProtection.js

```diff
@@ -0,0 +1,304 @@
+    }
+
+    // Farble the array and store the hash of this
+    // so that we don't farble the same data again.
+    farbleArrayData(channelData)
+    const hash = hashArray(channelData)
+    hashes.add(hash)
+
+    return channelData
+  }
+
+  // 2. Farble "destination" methods
+  const structuresToFarble = [
+    [window.AnalyserNode, 'getFloatFrequencyData'],
+    [window.AnalyserNode, 'getByteFrequencyData'],
+    [window.AnalyserNode, 'getByteTimeDomainData'],
+    [window.AnalyserNode, 'getFloatTimeDomainData']
+  ]
+
+  for (const [structure, methodName] of structuresToFarble) {
+    const origImplementation = structure.prototype[methodName]
+    structure.prototype[methodName] = function () {
+      Reflect.apply(origImplementation, this, arguments)
+      farbleArrayData(arguments[0])
+    }
+  }
+
+  // 3. Farble plugin data
+  // Function that create a fake mime-type based on the given fake data
+  const makeFakeMimeType = (fakeData) => {
+    return Object.create(window.MimeType.prototype, {
+      suffixes: {
+        get: function () {
+          return fakeData.suffixes
+        }
+      },
+      type: {
+        get: function () {
+          return fakeData.type
+        }
+      },
+      description: {
+        get: function () {
+          return fakeData.description
+        }
+      }
+    })
+  }
+
+  // Create a fake plugin given the plugin data
+  const makeFakePlugin = (pluginData) => {
+    const newPlugin = Object.create(window.Plugin.prototype, {
+      description: {
+        get: function () {
+          return pluginData.description
+        }
+      },
+      name: {
+        get: function () {
+          return pluginData.name
+        }
+      },
+      filename: {
+        get: function () {
+          return pluginData.filename
+        }
+      },
+      length: {
+        get: function () {
+          return pluginData.mimeTypes.length
+        }
+      }
+    })
+
+    // Create mime-types and link them to the new plugin
+    for (const [index, mimeType] of pluginData.mimeTypes.entries()) {
+      const newMimeType = makeFakeMimeType(mimeType)
+
+      newPlugin[index] = newMimeType
+      newPlugin[newMimeType.type] = newMimeType
+
+      Reflect.defineProperty(newMimeType, 'enabledPlugin', {
+        get: function () {
+          return newPlugin
+        }
+      })
+    }
+
+    // Fix .item() function to return the correct item
+    newPlugin.item = function (index) {
+      return newPlugin[index]
+    }
+
+    return newPlugin
+  }
+
+  // We need the original length so we can reference it (as we will change it)
+  const plugins = window.navigator.plugins
+  const originalPluginsLength = plugins.length
+  const originalItemFunction = plugins.item
+
+  // Adds a fake plugin for the given index on fakePluginData
+  const addPluginAtIndex = (newPlugin, index) => {
+    const pluginPosition = originalPluginsLength + index
+    window.navigator.plugins[pluginPosition] = newPlugin
```

Comment:
If `window.navigator.plugins` is ever null, this will break.
This API `window.navigator.plugins` is also deprecated and will be removed in the future.

Reply:
Should I do a `if (window.navigator.plugins)` check? 
Note. that this script is injected to a web view which uses web kit. So there is no need to do cross platform checks. 
And yes we're aware the script is deprecated.

Reply:
I added an `if (window.navigator.plugins !== undefined)` check in case apple decides to remove this API